src/main/java/com/googleinterns/zoomtube/data/Comment.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.googleinterns.zoomtube.data;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.repackaged.com.google.common.base.Optional;
+import com.google.auto.value.AutoValue;
+import java.util.Date;
+import javax.annotation.Nullable;
+
+/** Contains data related to a comment in a discussion. */
+@AutoValue
+public abstract class Comment {
+  public static final String ENTITY_KIND = "Comment";
+  public static final String PROP_LECTURE = "lecture";
+  public static final String PROP_PARENT = "parent";
+  public static final String PROP_TIMESTAMP = "timestamp";
+  public static final String PROP_AUTHOR = "author";
+  public static final String PROP_CONTENT = "content";
+  public static final String PROP_CREATED = "created";
+
+  public static Comment create(Key key, Key lecture, Optional<Key> parent, double timestamp,
```

Comment:
This takes as input an optional parent. I would create a Builder class so that you do not need to pass optional values in to build the Comment object. Please see: https://github.com/google/auto/blob/master/value/userguide/builders.md

Reply:
I realize you have fromEntity method. Does this create() method need to be public? If it is private then you wouldn't need to do what I recommended.

https://docs.google.com/document/d/1Ezd4q96yzgkHre_7VF44vxNiggKnyjVkgUpVcyfEnwA/edit
This document talks about avoiding to use booleans in public methods, but this is also applicable to Optional and null.

Reply:
For now it can be private as only fromEntity is used.  I'll make a builder if I ever need it

Reply:
done

Reply:
Great! Yea making methods which are not used private is best.